src/main/java/com/blamejared/crafttweaker/impl/recipes/handlers/vanilla/CookingRecipeHandler.java

```diff
@@ -0,0 +1,66 @@
+package com.blamejared.crafttweaker.impl.recipes.handlers.vanilla;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.IReplacementRule;
+import com.blamejared.crafttweaker.api.util.StringUtils;
+import com.blamejared.crafttweaker.impl.item.MCItemStackMutable;
+import com.google.common.collect.ImmutableMap;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.AbstractCookingRecipe;
+import net.minecraft.item.crafting.BlastingRecipe;
+import net.minecraft.item.crafting.CampfireCookingRecipe;
+import net.minecraft.item.crafting.FurnaceRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.SmokingRecipe;
+import net.minecraft.util.ResourceLocation;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@IRecipeHandler.For(BlastingRecipe.class)
+@IRecipeHandler.For(CampfireCookingRecipe.class)
+@IRecipeHandler.For(FurnaceRecipe.class)
+@IRecipeHandler.For(SmokingRecipe.class)
+public final class CookingRecipeHandler implements IRecipeHandler<AbstractCookingRecipe> {
+    @FunctionalInterface
+    private interface CookingRecipeFactory<T extends AbstractCookingRecipe> {
+        T create(final ResourceLocation id, final String group, final Ingredient ingredient, final ItemStack result, final float experience, final int cookTime);
+    }
+    
+    private static final Map<IRecipeType<?>, Pair<String, CookingRecipeFactory<?>>> LOOKUP = ImmutableMap
+            .<IRecipeType<?>, Pair<String, CookingRecipeFactory<?>>>builder()
+            .put(IRecipeType.BLASTING, Pair.of("blastFurnace", BlastingRecipe::new))
+            .put(IRecipeType.CAMPFIRE_COOKING, Pair.of("campfire", CampfireCookingRecipe::new))
+            .put(IRecipeType.SMELTING, Pair.of("furnace", FurnaceRecipe::new))
+            .put(IRecipeType.SMOKING, Pair.of("smoker", SmokingRecipe::new))
+            .build();
+    
+    @Override
+    public String dumpToCommandString(final IRecipeManager manager, final AbstractCookingRecipe recipe) {
+        
+        return String.format(
+                "%s.addRecipe(%s, %s, %s, %s, %s);",
+                LOOKUP.get(recipe.getType()).getFirst(),
+                StringUtils.quoteAndEscape(recipe.getId()),
+                new MCItemStackMutable(recipe.getRecipeOutput()).getCommandString(),
+                IIngredient.fromIngredient(recipe.getIngredients().get(0)),
+                recipe.getExperience(),
+                recipe.getCookTime()
+        );
```

Comment:
We had the discussion a while back:
Do we want a custom builder, since the conversions to the output format is in many cases
```java
return RecipeStringBuilder.forFormat("{manager}.addRecipe({name}, {output}, {input}, {experience}, {cookTime}")
    .set("manager", LOOKUP.get(recipe.getType()).getFirst())
    .quoteString("name", recipe.getId())
    .setItemStack("output", recipe.getRecipeOutput())
    .setIngredient("input", recipe.getIngredients().get(0)
    .set("experience", recipe.getExperience())
    .set("cookingTime", recipe.getCookTime())
```